cmake/BuildFunctions.cmake

```diff
@@ -98,7 +98,10 @@ function(create_test_executable)
   target_include_directories(${exec_name}
     PRIVATE ${GTEST_INCLUDE_DIRS})
 
-  target_link_libraries(${exec_name}
+  set_property(TARGET ${exec_name} 
+    APPEND_STRING PROPERTY COMPILE_FLAGS "-DBUILD_IS_TEST_ENVIRONMENT")
```

Comment:
When will this define be set? I don't know this property.

Reply:
it's set by `create_test_executable()`. It simply adds the define `BUILD_IS_TEST_ENVIRONMENT`. I guess it could also be set by `create_integration_test_executable()`, so we get the validation during integration tests.